### PR TITLE
chore: update job tests with retries

### LINKED ISSUE #6829: com.example.JobsIntegrationTests: generatesLogs failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: c13fea01ff731da279aba8541897c3db2fceb1e2
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/60125967-66c8-42fb-8912-487d47806e5a), [Sponge](http://sponge2/60125967-66c8-42fb-8912-487d47806e5a)
status: failed
<details><summary>Test output</summary><br><pre>java.lang.AssertionError: Log was not found.
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at com.example.JobsIntegrationTests.generatesLogs(JobsIntegrationTests.java:117)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
</pre></details>

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #6829

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java [hunk 1]
```diff
@@ -104,15 +104,22 @@ public void generatesLogs() throws Exception {
...
+        Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
+        for (LogEntry logEntry : entries.iterateAll()) {
+          if (!logEntry.getLogName().contains("cloudaudit")) {
+            Payload<String> payload = logEntry.getPayload();
+            if (payload.getData().contains("Task")) { 
+              found = true; 
+            }
           }
         }
+        if (found) {
```
Comment (Shabirmean) [inline run/jobs/src/test/java/com/example/JobsIntegrationTests.java:119]: Can we break right after line 115? Seems like `found` never changes anywhere else.

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java [hunk 2]
```diff
@@ -104,15 +104,22 @@ public void generatesLogs() throws Exception {
...
+          break;
+        }
+        Thread.sleep(i * 30000);
```
Comment (Shabirmean) [inline run/jobs/src/test/java/com/example/JobsIntegrationTests.java:122]: After the first iteration there will no sleeps due to `i=0`. So the first two iteration will run one after the other. Is this the behavior we want? Or should we pause after the first iteration as well?

**Review by Shabirmean — COMMENTED** [review]
I am just doing a `shadow-review`. Left some comments, I might not be fully aware of the background of the issue; so feel free to resolve them if they are not relevant.

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java (hunk 2 above)
Comment (kurtisvg) [inline run/jobs/src/test/java/com/example/JobsIntegrationTests.java:120]: nit: It's nicer to use the java 8 time APIs here
```suggestion
        TimeUnit.SECONDS.sleep(i * 30)
```

File: run/jobs/src/test/java/com/example/JobsIntegrationTests.java (hunk 1 above)
Comment (averikitsch) [reply run/jobs/src/test/java/com/example/JobsIntegrationTests.java]: updated

**Review by kurtisvg — CHANGES_REQUESTED** [review]

**Review by Shabirmean — APPROVED** [review]
LGTM! Thank you for the changes!

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**